bzt/engine.py

```diff
@@ -1072,29 +1072,45 @@ def __init__(self, config):
 
 
 class HTTPRequest(Request):
-    def __init__(self, config, engine):
-        super(HTTPRequest, self).__init__(config)
+    def __init__(self, config, scenario, engine):
         self.engine = engine
+        self.log = self.engine.log.getChild(self.__class__.__name__)
+        super(HTTPRequest, self).__init__(config)
+        self.scenario = scenario
         msg = "Option 'url' is mandatory for request but not found in %s" % config
-        self.url = config.get("url", TaurusConfigError(msg))
-        self.label = config.get("label", self.url)
-        self.method = config.get("method", "GET")
-        self.headers = config.get("headers", {})
-        self.timeout = config.get("timeout", None)
-        self.think_time = config.get("think-time", None)
-
-        body = config.get('body', None)
-        body_file = config.get('body-file', None)
+        self.url = self.config.get("url", TaurusConfigError(msg))
+        self.label = self.config.get("label", self.url)
+        self.method = self.config.get("method", "GET")
+
+        # TODO: add method to join dicts/lists from scenario/request level?
+        self.headers = self.config.get("headers", {})
+
+        self.keepalive = self.config.get('keepalive', None)
+        self.timeout = self.config.get('timeout', None)
+        self.think_time = self.config.get('think-time', None)
+        self.follow_redirects = self.config.get('follow-redirects', None)
+        self.body = self.__get_body()
+
+    def by_priority(self, name, default=None):
```

Comment:
Maybe we should rename it back to `priority_option`? `by_priority` looks nice and DSL-ish, but isn't in line with our usual method naming.